### PR TITLE
Add Demo link to landing page

### LINKED ISSUE #1041: Allow users to explore online demo from the landing page
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡
1. A 'Try now' type of CTA would be needed for users to quickly explore Toolpad. 
Ideally we should let users see the demo video and then show an option to try Toolpad, but then it will be too hidden. 

2. Add Google Analytics to this button.

### Example
https://usefathom.com/

### LINKED ISSUE #1132: Update demo video and docs screenshots
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Reminder task to be done before https://github.com/mui/mui-toolpad/issues/900

2. In the header nav bar, show Try demo and self-host CTAs. Also, directly link the Toolpad GitHub repo

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes #1041 
- Closes #1132 
- Visual demonstration 
Screenshot 2022-11-09 at 1 38 27 PM

- This is the deploy preview https://deploy-preview-1228--mui-toolpad-docs.netlify.app/toolpad/

### CODE REVIEW

File: docs/src/components/landing/Hero.js [hunk 1]
```diff
@@ -40,10 +41,20 @@ export default function Hero() {
           Drag and drop pre-built components, connect to data sources and APIs, and build your
           internal tools 10x faster. Open-source and powered by MUI.
         </Typography>
+        <Button
+          size="large"
+          variant="contained"
+          href="https://demo.toolpad.io/"
+          target="_blank"
```
Comment (apedroferreira) [inline docs/src/components/landing/Hero.js:48]: Maybe there's no need to open in a separate tab?
If the user tries the demo, do we want them to still use the landing page?

File: docs/src/components/landing/DemoVideo.js [hunk 2]
```diff
@@ -16,35 +16,48 @@ const Video = styled('video')(({ theme }) => ({
...
-        borderColor:
-          theme.palette.mode === 'dark' ? theme.palette.primaryDark[400] : theme.palette.grey[300],
-        borderRadius: '10px 10px 0 0',
-        borderWidth: '1px 1px 0 1px',
-      })}
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'center',
+        mt: { xs: '20px', sm: '60px' },
```
Comment (apedroferreira) [inline docs/src/components/landing/DemoVideo.js:23]: Would the units from MUI work for margin too? Or maybe we're trying to have more specific margin in this case.

File: docs/src/components/landing/CardGrid.js [hunk 3]
```diff
@@ -37,12 +37,16 @@ const cardContentRootStyle = (imageUrl) => ({
...
 export default function CardGrid(props) {
-  const { content } = props;
+  const { content, dark } = props;
   return (
     <Box
-      sx={{
-        bgcolor: (theme) => (theme.palette.mode === 'dark' ? 'primaryDark.900' : 'grey.50'),
-      }}
+      sx={
+        dark
```
Comment (apedroferreira) [inline docs/src/components/landing/CardGrid.js:44]: I don't understand the whole context here, but having a `dark` prop and then still checking for `theme.palette.mode === 'dark'` feels a bit weird?

File: docs/src/components/landing/DemoVideo.js [hunk 4]
```diff
@@ -16,35 +16,48 @@ const Video = styled('video')(({ theme }) => ({
...
-        loop
-        poster="/static/toolpad/marketing/index-hero-video-poster.jpg"
+      <Box
+        sx={(theme) => ({
+          borderStyle: 'solid',
+          maxWidth: 1100,
+          height: { xs: 200, sm: 390, md: 445, lg: 615 },
+          width: {
+            xs: 360,
+            sm: theme.breakpoints.values.sm + 100,
```
Comment (apedroferreira) [inline docs/src/components/landing/DemoVideo.js:35]: Could we use a constant for the difference between video width and breakpoint width?

File: docs/src/components/landing/DemoVideo.js [hunk 5]
```diff
@@ -16,35 +16,48 @@ const Video = styled('video')(({ theme }) => ({
...
-        sx={{
-          borderWidth: { xs: '10px 10px 0 10px', sm: '20px 20px 0 20px' },
-        }}
-        muted
-        autoPlay
-        playsInline
-        loop
-        poster="/static/toolpad/marketing/index-hero-video-poster.jpg"
+      <Box
+        sx={(theme) => ({
```
Comment (apedroferreira) [inline docs/src/components/landing/DemoVideo.js:29]: Lots of styles here, maybe we could have them in its own styled component for example?

**Review by apedroferreira — COMMENTED** [review]
Some of my comments may make more sense than others.
Also I just looked at the code, not sure how to take a look at a preview of the landing page with these changes?

File: docs/src/components/landing/Hero.js (hunk 1 above)
Comment (bharatkashyap) [reply docs/src/components/landing/Hero.js:48]: I think the idea is that the user goes to the demo, returns to the landing page and then signs up for the beta if they liked the demo. Which is why a separate tab.

File: docs/src/components/landing/DemoVideo.js (hunk 2 above)
Comment (bharatkashyap) [reply docs/src/components/landing/DemoVideo.js:23]: I could use `theme.spacing` to get the values instead

File: docs/src/components/landing/CardGrid.js (hunk 3 above)
Comment (bharatkashyap) [reply docs/src/components/landing/CardGrid.js:44]: The `dark` prop is to specify if the grid should have a darker background relative to the other sections on the landing page; agreed that the naming is confusing. 

I'll rename the prop to `darker`

**Comment by bharatkashyap** [comment]
> Some of my comments may make more sense than others. Also I just looked at the code, not sure how to take a look at a preview of the landing page with these changes?

This is the deploy preview https://deploy-preview-1228--mui-toolpad-docs.netlify.app/toolpad/

File: docs/src/components/landing/Hero.js (hunk 1 above)
Comment (apedroferreira) [reply docs/src/components/landing/Hero.js:48]: Ok, sounds good in that case.

**Review by gerdadesign — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**